test/e2e/e2e.go

```diff
@@ -46,3 +51,41 @@ func CreateRouteAndConfig(t *testing.T, clients *test.Clients, image string, opt
 	_, err := test.CreateRoute(t, clients, names)
 	return names, err
 }
+
+// WaitForScaleToZero will wait for the deployment specified by names to scale to
+// 0 replicas. Will wait up to 6 times the configured ScaleToZeroGracePeriod before failing.
+func WaitForScaleToZero(t *testing.T, deploymentName string, clients *test.Clients) error {
+	t.Helper()
+
+	t.Logf("Waiting for %q to scale to zero", deploymentName)
+
+	return pkgTest.WaitForDeploymentState(
+		clients.KubeClient,
+		deploymentName,
+		func(d *v1beta1.Deployment) (bool, error) {
+			t.Logf("Deployment %q has %d replicas", deploymentName, d.Status.ReadyReplicas)
+			return d.Status.ReadyReplicas == 0, nil
+		},
+		"DeploymentIsScaledDown",
+		test.ServingNamespace,
+		scaleToZeroGracePeriod(t, clients.KubeClient),
+	)
+}
+
+func scaleToZeroGracePeriod(t *testing.T, client *pkgTest.KubeClient) time.Duration {
+	t.Helper()
+
+	autoscalerCM, err := client.Kube.CoreV1().ConfigMaps("knative-serving").Get("config-autoscaler", metav1.GetOptions{})
+	if err != nil {
+		t.Log("failed to Get autoscaler configmap, falling back to DefaultScaleToZeroGracePeriod")
```

Comment:
```suggestion
		t.Logf("Failed to Get autoscaler configmap = %v, falling back to DefaultScaleToZeroGracePeriod", err)
```
error might be useful.
Also the logs, unlike errors, are statements, so normal capitalization applies.